app/controllers/openid_connect/authorization_controller.rb

```diff
@@ -23,6 +23,12 @@ def index
       return redirect_to_account_or_verify_profile_url if profile_or_identity_needs_verification?
       return redirect_to(sign_up_completed_url) if needs_completion_screen_reason
       link_identity_to_service_provider
+
+      result = @authorize_form.submit
```

Comment:
Would it be worth changing this up so that `pre_validate_authorize_form` is either merged into this method or called from the method, such that the same result could be shared by both?

I don't think the logic for validating is especially heavy, but could be worth optimizing since I imagine it's a hot path, and there's quite a few validations in `OpenidConnectAuthorizeForm`.

Reply:
The problem is, we need to run `link_identity_to_service_provider`, but only if the the form is successful and so if we want to get the right `code` in the result payload from the form, we need to run that after it links... so that's why I went with this double submit/sandwich approach.

I think a right thing to do would be to fund a way to untangle, but I feel like given time constraints, now is not that time? I did file [LG-7544](https://cm-jira.usa.gov/browse/LG-7544) so we could keep track of this